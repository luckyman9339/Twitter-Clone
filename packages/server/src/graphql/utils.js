export const decodeCursor = cursor => {
  const content = new Buffer(cursor, 'base64').toString('binary');
  const [after, order] = content.split(';');
  return { after: new Date(after), order };
};

export const encodeCursor = ({ after, order }) => {
  const input = [new Date(after).toISOString(), order].join(';');
  return Buffer.from(input).toString('base64');
};

export const buildEdgesForTweetsPagination = (tweets, order) =>
  tweets.map(tweet => {
    const type = tweet.retweetForTweetId
      ? 'RETWEET'
      : tweet.likeForTweetId
      ? 'LIKE'
      : 'TWEET';

    const node = { ...tweet, type };

    return {
      cursor: encodeCursor({
        after: tweet.created_at,
        order,
      }),
      node,
    };
  });
