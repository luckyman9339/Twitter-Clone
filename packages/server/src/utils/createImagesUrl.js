module.exports = {
  createUserAvatarUrl: email => {
    return `https://api.adorable.io/avatars/200/${email}.png`;
  },
};
