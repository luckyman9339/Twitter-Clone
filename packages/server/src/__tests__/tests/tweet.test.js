import knexCleaner from 'knex-cleaner';
import { request } from '../utils';
import knex from 'db/knex';

let user = null;

beforeAll(async () => {
  await knexCleaner.clean(knex, {
    ignoreTables: ['knex_migrations', 'knex_migrations_lock'],
  });
  await knex.seed.run();
  user = (await knex.table('users').where({ username: 'test_user' }))[0];
});

it('should fetch a tweet', async () => {
  const query = `
    query GetTweet($tweetId: ID!) {
      tweet(id: $tweetId) {
        id
        content
        retweetCount
        likeCount
        retweeted
        liked
        user {
          id
          name
          username
        }
      }
    }
  `;

  const variables = { tweetId: '53c1c783-bebf-4f62-bd46-766c827e7d62' };

  const result = await request(query, { variables });
  expect(result).toMatchSnapshot();
});

it('should create a tweet', async () => {
  const context = { user };

  const query = `
    mutation CreateTweet($input: CreateTweetInput!) {
      createTweet(input: $input) {
        content
        user {
          id
          username
        }
      }
    }
  `;

  const variables = {
    input: {
      content: 'Hello Wold var',
    },
  };

  const result = await request(query, { context, variables });
  expect(result).toMatchSnapshot();
});
