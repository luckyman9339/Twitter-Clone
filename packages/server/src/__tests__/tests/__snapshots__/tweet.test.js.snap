// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a tweet 1`] = `
Object {
  "data": Object {
    "createTweet": Object {
      "content": "Hello Wold var",
      "user": Object {
        "id": "551cf7db-df68-4ad5-af6c-1935759b9291",
        "username": "test_user",
      },
    },
  },
}
`;

exports[`should fetch a tweet 1`] = `
Object {
  "data": Object {
    "tweet": Object {
      "content": "Hello World from User One!",
      "id": "53c1c783-bebf-4f62-bd46-766c827e7d62",
      "likeCount": 0,
      "liked": false,
      "retweetCount": 0,
      "retweeted": false,
      "user": Object {
        "id": "551cf7db-df68-4ad5-af6c-1935759b9291",
        "name": "Test User One",
        "username": "test_user",
      },
    },
  },
}
`;
